AAT_gae/src/main/java/com/aat/restlet/QRCodeResource.java

```diff
@@ -0,0 +1,85 @@
+package com.aat.restlet;
+
+import java.security.SecureRandom;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+import java.util.HashMap;
+import java.util.List;
+
+import org.restlet.representation.Representation;
+import org.restlet.representation.StringRepresentation;
+import org.restlet.resource.Get;
+import org.restlet.resource.ServerResource;
+
+import com.aat.datastore.AttendanceRecord;
+import com.aat.datastore.Student;
+import com.aat.datastore.User;
+import com.googlecode.objectify.ObjectifyService;
+import com.googlecode.objectify.Ref;
+
+
+
+/**
+ * Creates a token base	d on Student id, group id and the date of first the day(Sunday) of the current week. 
+ * */
+public class QRCodeResource extends ServerResource {
+	
+	@Get
+	public Representation retrieve(){
+							
+		String userID = getAttribute("user_id");
+		String groupID = getAttribute("group_id");
+		StringBuffer sBufferToken = new StringBuffer();
+		Calendar calendar = GregorianCalendar.getInstance();
+		calendar.setFirstDayOfWeek(Calendar.SUNDAY);;
+		calendar.set(Calendar.DAY_OF_WEEK, Calendar.SUNDAY);
+		SimpleDateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy");
+		String date = dateFormat.format(calendar.getTime());
+		
+		HashMap<String,String> token = getTokens(userID, groupID, date);
+		
+		if (token.get(date) == null){
+			SecureRandom random = new SecureRandom();
+			byte bytes[] = new byte[20];
+			random.nextBytes(bytes);
+			token.put(date, bytes.toString());
```

Comment:
Don't you have to save this back to the AttendenceRecord using ofy().save()

Otherwise, I don't see the tokens ever being saved to the AttendanceRecord stored in the datastore. As a result, a new token will be created every time.